### PR TITLE
test: update migration test case, add single tab app and bot

### PR DESCRIPTION
**Pull request by wenytang-ms**

### CODE REVIEW

File: packages/fx-core/tests/core/hooks.test.ts [hunk 1]
```diff
@@ -63,6 +63,7 @@ import {
 } from "./utils";
 import * as dotenv from "dotenv";
 let mockedEnvRestore: () => void;
+const fileEncoding = "UTF8";
```
Comment (xzf0587) [inline packages/fx-core/tests/core/hooks.test.ts:66]: There is a const string ConstantString.UTF8Encoding defines the "UTF8" already.

File: packages/fx-core/tests/core/hooks.test.ts [hunk 2]
```diff
@@ -576,7 +568,164 @@ describe("Middleware - others", () => {
...
+          path.join(projectPath, "templates", "azure", "provision", "frontendHosting.bicep"),
+          fileEncoding
+        ),
+        await fs.readFile(
+          path.join(botBicepFilePath, "frontendProvision.result.bicep"),
+          fileEncoding
+        )
+      );
+    });
+    it("successfully migration arm templates only tab", async () => {
```
Comment (xzf0587) [inline packages/fx-core/tests/core/hooks.test.ts:643]: here is a bot test, not a tab?

File: packages/fx-core/tests/core/hooks.test.ts [hunk 3]
```diff
@@ -576,7 +568,164 @@ describe("Middleware - others", () => {
...
+        await fs.readFile(
+          path.join(projectPath, "templates", "azure", "provision", "identity.bicep"),
+          fileEncoding
+        ),
+        await fs.readFile(
+          path.join(identityBicepFilePath, "identityProvision.result.bicep"),
+          fileEncoding
+        )
+      );
+      const botBicepFilePath = path.join(
```
[the remark this answers is not in the document]
Comment (wenytang-ms) [reply packages/fx-core/tests/core/hooks.test.ts:623]: thanks, is frontendBicepFilePath

File: packages/fx-core/tests/core/hooks.test.ts (hunk 2 above)
Comment (wenytang-ms) [reply packages/fx-core/tests/core/hooks.test.ts:643]: update, a bot test, thanks

**Review by xzf0587 — APPROVED** [review]

**Pull request merged**